src/Dumplie/Application/Metadata/HydrationData.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Application\Metadata;
+
+interface HydrationData
+{
+    /**
+     * @param Storage $storage
+     *
+     * @return array
+     */
+    public function produce(Storage $storage) : array;
```

Comment:
I'm not sure if putting storage as a dependency here is good decision. Most kind of data won't use it anyway and associations can get storage through constructor